src/AppInstallerRepositoryCore/Microsoft/SQLiteIndex.h

```diff
@@ -57,7 +58,7 @@ namespace AppInstaller::Repository::Microsoft
         };
 
         // Opens an existing index database.
-        static SQLiteIndex Open(const std::string& filePath, OpenDisposition disposition);
+        static SQLiteIndex Open(const std::string& filePath, OpenDisposition disposition, TempSQLiteIndexFile&& tempIndexFileHandle = {});

```

Comment:
I don't love this being passed in, but I understand the need.  Can we change the type to `WriteLockedFile` or something more generic?  That could be used for both the package and index files (with a `create` bool that controls both open and remove on destruction functionality).

Then also just move the "extract the index file" part out of the type as it isn't really necessary to be included.